runtime/lua/vim/lsp.lua

```diff
@@ -571,6 +571,11 @@ local function lsp_enable_callback(bufnr)
           end)
         end)
       else
+        if vim.go.autochdir == true then
+          -- update directory before resolving root markers
+          -- setting value triggers autochdir callback
+          vim.go.autochdir = true
```

Comment:
how does this actually fix the issue?

Reply:
By triggering `do_autochdir` when the option is set:
https://github.com/neovim/neovim/blob/0d658660c29e920e74c4dade3819d80dccad0dde/src/nvim/option.c#L1951